src/doc/trpl/glossary.md

```diff
@@ -37,3 +25,41 @@ And `2 + (3 * 4)` would look like this:
    / \
   3   4
 ```
+
+### Arity
+
+Arity refers to the number of arguments a function or operation takes.
+
+```rust
+let x = (2, 3);
+let y = (4, 6);
+let z = (8, 2, 6);
+```
+
+In the example above `x` and `y` have arity 2. `z` has arity 3.
+
+### Expression
+
+In computer programming, an expression is a combination of values, constants,
+variables, operators and functions that evaluate to a single value. For example,
+`2 + (3 * 4)` is an expression that returns the value 14. It is worth noting
+that expressions can have side-effects. For example, a function included in an
+expression might perform actions other than simply returning a value.
+
+### Expression-Oriented Language
+
+In early programming languages [expressions][expression] and
+[statements][statement] were two separate syntactic categories: expressions had
+a value and statements did things. However, later languages blurred this
+distinction, allowing expressions to do things and statements to have a value.
```

Comment:
Wish I could provide a citation. I didn't know what expression-oriented languages were and hoped the glossary would help. In the end I read a number of web pages (including Wikipedia!) and my description was the best I could put together. If there are inaccuracies, I just hope someone in the know corrects them!



> On 25 Jul 2015, at 03:39, Tshepang Lekhonkhobe <notifications@github.com> wrote:
> 
> In src/doc/trpl/glossary.md:
> 
> > +In the example above `x` and `y` have arity 2. `z` has arity 3.
> > +
> > +### Expression
> > +
> > +In computer programming, an expression is a combination of values, constants,
> > +variables, operators and functions that evaluate to a single value. For example,
> > +`2 + (3 * 4)` is an expression that returns the value 14. It is worth noting
> > +that expressions can have side-effects. For example, a function included in an
> > +expression might perform actions other than simply returning a value.
> > +
> > +### Expression-Oriented Language
> > +
> > +In early programming languages [expressions][expression] and
> > +[statements][statement] were two separate syntactic categories: expressions had
> > +a value and statements did things. However, later languages blurred this
> > +distinction, allowing expressions to do things and statements to have a value.
> Do you have a citation for the claim about early vs. later programming languages? Alternatively, can you just give examples of those languages instead?
> 
> —
> Reply to this email directly or view it on GitHub.
>